File: 2042D-recommendations.js
function findStronglyRecommendedTracks(t, testCases) {
    let results = [];
    
    for (let caseIndex = 0; caseIndex < t; caseIndex++) {
        let [n, segments] = testCases[caseIndex];
        let recommendedCounts = new Array(n).fill(0);
        let usersSegments = segments.map(([l, r]) => [l, r]);

        for (let i = 0; i < n; i++) {
            let [li, ri] = usersSegments[i];
            let predictors = [];

            for (let j = 0; j < n; j++) {
                if (i !== j) {
                    let [lj, rj] = usersSegments[j];
                    if (lj <= li && ri <= rj) {
                        predictors.push([lj, rj]);
                    }
                }
            }

            if (predictors.length === 0) {
                recommendedCounts[i] = 0;
                continue;
            }

            let intersectionL = Math.max(...predictors.map(([lj]) => lj));
            let intersectionR = Math.min(...predictors.map(([_, rj]) => rj));
            
            if (intersectionL <= intersectionR) {
                let totalRecommended = (intersectionR - intersectionL + 1) - Math.max(0, ri - li + 1);
                recommendedCounts[i] = totalRecommended;
            } else {
                recommendedCounts[i] = 0;
            }
        }

        results.push(recommendedCounts);
    }

    return results;
}

function main() {
    const t = parseInt(prompt("Enter number of test cases: "), 10);
    let testCases = [];

    for (let i = 0; i < t; i++) {
        const n = parseInt(prompt("Enter number of segments: "), 10);
        let segments = [];
        
        for (let j = 0; j < n; j++) {
            const input = prompt("Enter segment (l r): ").split(" ").map(Number);
            segments.push([input[0], input[1]]);
        }
        
        testCases.push([n, segments]);
    }

    let results = findStronglyRecommendedTracks(t, testCases);
    
    results.forEach(result => console.log(result.join("\n")));
}

main();

File: 2042D-recommendations.py
# Author : AlifSrSE
# Date : 2025-03-03
# Problem link : https://codeforces.com/contest/2042/problem/D

import sys

def find_strongly_recommended_tracks(test_cases):
    results = []
    
    for n, segments in test_cases:
        recommended_counts = [0] * n
        
        for i in range(n):
            li, ri = segments[i]
            min_r = float('inf')
            max_l = -float('inf')
            predictor_count = 0
            
            for j in range(n):
                if i != j:
                    lj, rj = segments[j]
                    if lj <= li and ri <= rj:
                        max_l = max(max_l, lj)
                        min_r = min(min_r, rj)
                        predictor_count += 1
            
            if predictor_count == 0:
                continue
            
            if max_l <= min_r:
                recommended_counts[i] = max(0, min_r - max_l + 1 - (ri - li + 1))
        
        results.append(recommended_counts)
    
    return results

def main():
    t = int(sys.stdin.readline())
    test_cases = []
    
    for _ in range(t):
        n = int(sys.stdin.readline())
        segments = []
        for _ in range(n):
            l, r = map(int, sys.stdin.readline().split())
            segments.append((l, r))
        test_cases.append((n, segments))
    
    results = find_strongly_recommended_tracks(test_cases)
    
    output = sys.stdout.write
    for result in results:
        output('\n'.join(map(str, result)) + '\n')

if __name__ == "__main__":
    main()